### PR TITLE
Fix #65: Missing Notifications

### LINKED ISSUE #65: Missing Notifications
In ClientBleGattCallback:

```kotlin
private val _event = MutableSharedFlow<ClientGattEvent>(
   extraBufferCapacity = 10, //Warning: because of this parameter we can miss notifications
   onBufferOverflow = BufferOverflow.DROP_OLDEST
)
```

The buffer size is way too small, and does indeed cause notification drops, as the warning says. Tested the same with 100 and I was no longer missing notifications.

### PR DESCRIPTION
**Pull request by sylwester-zielinski**
Fix #65: Missing Notifications

### CODE REVIEW

**Review by himalia416 — APPROVED** [review]

File: client/src/main/java/no/nordicsemi/android/kotlin/ble/client/main/service/ClientBleGattCharacteristic.kt [hunk 1]
```diff
@@ -207,16 +233,27 @@ class ClientBleGattCharacteristic internal constructor(
         mutex.lock()
         val stacktrace = Exception() //Helper exception to display valid stacktrace.
         return suspendCoroutine { continuation ->
-            logger.log(Log.DEBUG, "Write to characteristic - start, uuid: $uuid, value: $value, type: $writeType")
+            logger.log(
+                Log.DEBUG,
+                "Write to characteristic - start, uuid: $uuid, value: $value, type: $writeType"
```
Comment (philips77) [inline client/src/main/java/no/nordicsemi/android/kotlin/ble/client/main/service/ClientBleGattCharacteristic.kt:238]: Is the value here printed as hex or an if of a reference?

File: client/src/main/java/no/nordicsemi/android/kotlin/ble/client/main/service/ClientBleGattCharacteristic.kt [hunk 2]
```diff
@@ -237,7 +274,10 @@ class ClientBleGattCharacteristic internal constructor(
      */
     @RequiresPermission(Manifest.permission.BLUETOOTH_CONNECT)
     suspend fun splitWrite(value: DataByteArray, writeType: BleWriteType = BleWriteType.DEFAULT) {
-        logger.log(Log.DEBUG, "Split write to characteristic - start, uuid: $uuid, value: ${value}, type: $writeType")
+        logger.log(
+            Log.DEBUG,
+            "Split write to characteristic - start, uuid: $uuid, value: ${value}, type: $writeType"
```
Comment (philips77) [inline client/src/main/java/no/nordicsemi/android/kotlin/ble/client/main/service/ClientBleGattCharacteristic.kt:279]: Redundant brackets

File: client/src/main/java/no/nordicsemi/android/kotlin/ble/client/main/service/ClientBleGattCharacteristic.kt (hunk 1 above)
Comment (sylwester-zielinski) [reply client/src/main/java/no/nordicsemi/android/kotlin/ble/client/main/service/ClientBleGattCharacteristic.kt:238]: value is of type DataByteArray which overrides toString() and uses hex as an output

File: client/src/main/java/no/nordicsemi/android/kotlin/ble/client/main/service/ClientBleGattCharacteristic.kt (hunk 2 above)
Comment (sylwester-zielinski) [reply client/src/main/java/no/nordicsemi/android/kotlin/ble/client/main/service/ClientBleGattCharacteristic.kt]: Done

**Pull request merged**